index.rst

```diff
@@ -2,24 +2,180 @@
+There are also minimal functionality tests and other categories of tests used on different stages of the system lifecycle that will not be covered here.
+
+SAL Scripts
+-----------
+
+SAL Scripts are specialized components designed to execute short lived actions in the observatory control system, including, but not limited to; enabling components, preparing the system for calibrations, taking calibration, tracking targets and taking data, etc.
+In principle, most operations executed by the system will be done using the ScriptQueue, by SAL Scripts.
+
+By default, these components are developed using the BaseScript Python class provided by `salobj`_.
+They are developed in one of the two available repositories; ts_externalscripts and ts_standardscripts, should follow TSSW development guidelines, have unit testing and so on.
+See :tstn:`010` for details on the use case for each of these repositories.
+
+.. _salobj: ts-salobj.lsst.io
+.. _ts_externalscripts: https://github.com/lsst-ts/ts_externalscripts
+.. _ts_standardscripts: https://github.com/lsst-ts/ts_standardscripts
+
+The usage of SAL Scripts for integration testing comes, mainly, in two different varieties; regular operations that can be run in special conditions for testing purposes and custom made to test a particular condition.
+In the first case, we simply use a regular operation (e.g. slew and take data) with a particular configuration for which we can easily predict the outcome, whereas scripts designed to execute an integration tests have pre-defined predictable behavior.
+The first line of testing consist in making sure the script executes successfully, followed by a verification that the data produced by the different components matches our expectations (see :ref:`verification-artifacts` furthermore).
+
+The most obvious advantage of using regular operations for integration testing is that we are developing and using SAL Script that will be regularly used.
+Most of these operations exercise the vast majority of the system requirements, which makes them a perfect tool to verify them.
+Since the majority of these uses TSSW high-level libraries (e.g. ts_observatory_control), they are extremely maintainable, requiring minimal to no updates when there are underlying changes to the CSCs behavior.
+
+Jupyter Notebooks
+-----------------
+
+Jupyter notebooks provides a complimentary interactive interface into the Rubin Observatory Control system.
+They are used in parallel with SAL Scripts to execute tasks that require some user interaction or a finer degree of control over the operations.
+
+Although they can be very useful in the early development stages of the process, we do not recommend storing any long-term use software in notebooks.
+The main reason is that it is currently note possible to unit test Jupyter notebooks, which can leave to bit rot in short timescales, especially with the rapidly evolving software development one would expect in early commissioning stages.
```

Comment:
The primay reason is because it is not currently possible to unit test Jupyter notebooks under the TSSW CI framework,